src/components/account/login.vue

```diff
@@ -69,10 +79,63 @@ export default {
       password: ''
     };
   },
+  computed: {
+    oidcLoginPath() {
+      const { query } = this.$route;
+      const next = typeof query.next === 'string' ? query.next : null;
+      const qs = queryString({ next });
+      return `/v1/oidc/login${qs}`;
+    }
+  },
+  created() {
+    this.handleOIDCError();
+  },
   mounted() {
-    this.$refs.email.focus();
+    if (this.config.oidcEnabled)
+      window.addEventListener('pageshow', this.reenableIfPersisted);
+    else
+      this.$refs.email.focus();
+  },
+  beforeUnmount() {
+    if (this.config.oidcEnabled)
+      window.removeEventListener('pageshow', this.reenableIfPersisted);
   },
   methods: {
+    verifyNewSession() {
+      const sessionExpires = localStore.getItem('sessionExpires');
+      const newSession = sessionExpires == null ||
+        Number.parseInt(sessionExpires, 10) < Date.now();
+      if (!newSession) this.alert.info(this.$t('alert.alreadyLoggedIn'));
+      return newSession;
+    },
+    loginByOIDC(event) {
+      if (!this.verifyNewSession()) {
+        event.preventDefault();
+        return;
+      }
+
+      this.disabled = true;
+    },
+    handleOIDCError() {
+      const { oidcError } = this.$route.query;
+      if (this.config.oidcEnabled && typeof oidcError === 'string' &&
```

Comment:
`typeof oidcError === 'string'` looks unnecessary to me

Reply:
`this.$route.query.oidcError` can also be an array (if the query parameter is specified twice) or `null` (if it's specified without a value, i.e., `?oidcError`). We're about to insert the value into a template string before passing the result to `$te()`, so I just want to make sure that nothing breaks if something is wrong with the query parameter.